toolkit/offchain/src/reserve/deposit.rs

```diff
@@ -99,24 +99,23 @@ pub async fn deposit_to_reserve<
 	Ok(McTxHash(tx_id))
 }
 
-async fn get_utxo_with_tokens<T: QueryLedgerState>(
+pub(crate) async fn get_utxo_with_tokens<T: QueryLedgerState>(
 	scripts: &ReserveScripts,
 	token_id: &AssetId,
 	ctx: &TransactionContext,
 	client: &T,
-) -> Result<Option<OgmiosUtxo>, anyhow::Error> {
+) -> Result<Option<(OgmiosUtxo, ReserveDatum)>, anyhow::Error> {
 	let validator_address = scripts.validator.address_bech32(ctx.network)?;
 	let utxos = client.query_utxos(&[validator_address.clone()]).await?;
 	Ok(utxos
 		.into_iter()
-		.find(|utxo| {
-			utxo.value.native_tokens.contains_key(&scripts.auth_policy.script_hash())
-				&& utxo.datum.clone().is_some_and(|datum| {
-					decode_reserve_datum(datum).is_some_and(|reserve_datum| {
-						reserve_datum.immutable_settings.token == *token_id
-					})
-				})
+		.filter(|utxo| utxo.value.native_tokens.contains_key(&scripts.auth_policy.script_hash()))
+		.flat_map(|utxo| {
+			utxo.datum
+				.clone()
+				.and_then(|datum| decode_reserve_datum(datum).map(|datum| (utxo, datum)))
 		})
+		.find(|(_utxo, reserve_datum)| reserve_datum.immutable_settings.token == *token_id)
```

Comment:
Is there any change in behavior or only refactor? I cannot see difference. Anyway, now there is `ReserveData::get(genesis_utxo, &ctx, client).await?` and later `get_reserve_utxo` that should be used.
It was merged recently.